mage_ai/data_preparation/logging/logger_manager.py

```diff
@@ -48,19 +51,36 @@ def __init__(
         )
         self.stream = None
         if not self.logger.handlers:
-            if self.logging_config.destination_config:
-                handler = self.create_stream_handler()
-            else:
-                log_filepath = self.get_log_filepath(create_dir=True)
-                handler = logging.handlers.RotatingFileHandler(
-                    log_filepath,
-                    backupCount=10,
-                    maxBytes=MAX_LOG_FILE_SIZE,
-                )
+            try:
+                if self.logging_config.destination_config:
+                    handler = self.create_stream_handler()
+                else:
+                    log_filepath = self.get_log_filepath(create_dir=create_dir)
+                    handler = logging.handlers.RotatingFileHandler(
+                        log_filepath,
+                        backupCount=10,
+                        maxBytes=MAX_LOG_FILE_SIZE,
+                    )
+                    # TODO: Use TimedRotatingFileHandler for streaming pipelines
```

Comment:
The TimedRotatingFileHandler did write logs to separate files in separate directories based on time (every hour), but there was some odd behavior with the file being rotated and stored in the incorrect hour directory. This issue seems more relevant for streaming pipelines. Will revisit later in separate PR.